packages/algorand/src/htlc/index.ts

```diff
@@ -0,0 +1,149 @@
+        recipientAddress: string,
+        refundAddress: string,
+        hashLock: string,
+        expiration: number = this.config.expiration,
+        metadata: any
+    ) {
+        try {
+            const params = await this.provider.getTransactionParams();
+            const hashFn = 'sha256';
+            const htlc = new this.htlcTemplate.HTLC(refundAddress, recipientAddress, hashFn, hashLock, expiration, this.config.maxFee);
+
+            return fundHTLCContract(params, htlc, this.wallet, value, this.provider, metadata);
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    public async withdraw(
+        initiationTxHash: string,
+        recipientAddress: string,
+        refundAddress: string,
+        expiration: number,
+        secret: string,
+        metadata: any,
+        secretHash?: any
+    ) {
+        try {
+            const params = await this.provider.getTransactionParams();
+            const hashFn = 'sha256';
+            if (!secretHash) {
+                secretHash = sha256(secret);
+            }
+
+            const htlc = new this.htlcTemplate.HTLC(
+                refundAddress,
+                recipientAddress,
+                hashFn,
+                secretHash,
+                expiration,
+                this.config.maxFee
+            );
+
+            const txn = {
+                from: htlc.getAddress(),
+                to: Config().zeroAddress,
+                fee: 1,
+                type: 'pay',
+                amount: 0,
+                firstRound: params.firstRound,
+                lastRound: params.lastRound,
+                genesisID: params.genesisID,
+                genesisHash: params.genesisHash,
+                closeRemainderTo: recipientAddress,
+                note: formatNote(metadata),
+            };
+            const args = [];
+            args.push(secret);
+            const lsig = algosdk.makeLogicSig(htlc.getProgram(), args);
+            const rawSignedTxn = algosdk.signLogicSigTransaction(txn, lsig);
+
+            let tx = (await this.provider.sendRawTransaction(rawSignedTxn.blob, metadata));
+            return tx;
+        } catch (error) {
+            console.log('Error withdrawing', error);
+        }
+    }
+
+    async refund(
+        initiationTxHash: string,
+        recipientAddress: string,
+        refundAddress: string,
+        expiration: number,
+        secretHash: any,
+        metadata: RefundEvent) {
+        try {
+            const params = await this.provider.getTransactionParams();
+            const hashFn = 'sha256';
+
+            const htlc = new this.htlcTemplate.HTLC(
+                refundAddress,
+                recipientAddress,
+                hashFn,
+                secretHash,
+                expiration,
+                this.config.maxFee
+            );
+
+            const txn = {
+                from: htlc.getAddress(),
+                to: Config().zeroAddress,
```

Comment:
use `this.config`

Reply:
I prefer to use the zeroAddress as constant

Reply:
Then remove zeroAddress from the config and put it in the file that is used instead.